squashfs-tools/xz_wrapper.c

```diff
@@ -140,6 +148,86 @@ static int xz_options(char *argv[], int argc)
 			}
 		}
 
+		return 1;
+	} else if(strcmp(argv[0], "-Xpreset") == 0) {
+		char *b;
+		long val;
+
+		if(argc < 2) {
+			fprintf(stderr, "xz: -Xpreset missing preset-level (valid value 0-9)\n");
+			goto failed;
+		}
+
+		val = strtol(argv[1], &b, 10);
```

Comment:
Why don't you use atoi() since you are not going to use `b`?

Reply:
Hi, I'm reading around that it looks like atoi is scheduled to be deprecated and looks to be a common practice to use strtol due to better error handling. I'm changing it to setting NULL to b (effectively doing a atoi). Hope it's good for you. (this comes from checking actual difference between the 2 and reading this https://stackoverflow.com/a/60947439

Reply:
`strtol()` does have better error handling than `atoi()`. Optional: given we are using `strtol()` for better error handling, we might as well actually check that `b` is at the end of the string, so that we catch invalid arguments like `-Xpreset a1` or `-Xpreset 1b`.

Reply:
@wychen may be overkill but

```
		char *b;
		long val;

		if(argc < 2) {
			fprintf(stderr, "xz: -Xpreset missing preset-level (valid value 0-9)\n");
			goto failed;
		}

		val = strtol(argv[1], &b, 10);
		if (*b != '\0' || (int) val < 0 || (int) val & ~LZMA_PRESET_LEVEL_MASK) {
			fprintf(stderr, "xz: -Xpreset can't be "
				"negative or more than the max preset\n");
			goto failed;
		}
```

(and I guess reintroduce b in the other options with the same check?)

Reply:
This looks good. Thanks!